dig.go

```diff
@@ -259,7 +274,7 @@ func (c *Container) createInObject(t reflect.Type) (reflect.Value, error) {
 			return dest, err
 		}
 
-		v, err := c.get(f.Type)
+		v, err := c.get(edge{key: key{t: f.Type, name: f.Tag.Get(_nameTag)}})
```

Comment:
`edge{..., optional: isOptional}`? currently the lines below are applying the optional property.